### PR TITLE
Project description markdown functionality

### PR DESCRIPTION
**Pull request by kaja-osojnik**
Add react-markdown library to the project in order to be able to style project description in the description.yaml files.

Screenshots:
Before:
Screenshot 2024-01-15 at 16 01 36

After:
Screenshot 2024-01-15 at 16 01 09

### CODE REVIEW

**Review by matevz — CHANGES_REQUESTED** [review]
Thanks, this looks awesome!

Some nits:
1. Please fix styling for the [links](https://playground.oasis.io) (underscore, color...)
2. Can you make them open in a new window?
3. Please fix styling of the `preformatted values` to use oasis blue instead of red.

Current screenshot:
image

**Review by matevz — APPROVED** [review]
I got an error while building
```
$ yarn build
yarn run v1.22.5
$ tsc && vite build
src/components/ProjectListItem.tsx:20:7 - error TS6133: 'truncateText' is declared but its value is never read.

20 const truncateText = (text: string, maxLength: number) => {
         ~~~~~~~~~~~~


Found 1 error in src/components/ProjectListItem.tsx:20
```

Otherwise LGTM!

File: src/components/ProjectDialog.tsx [hunk 1]
```diff
@@ -80,13 +82,13 @@ const ProjectDialog: React.FC<ProjectDialogProps> = ({
...
-                      variant='body1'
-                      sx={{color: '#445E77', letterSpacing: '-0.5px', marginBottom: '24px'}}
-                      >
-                      <ReactMarkdown
-                      >{project.description}</ReactMarkdown>  
-                      </Typography>
+                      <Box
+                          sx={{color: '#445E77', letterSpacing: '-0.5px', marginBottom: '24px', fontFamily: "'Figtree Variable',sans-serif"}}
+                          >
+                      <ReactMarkdown>
```
Comment (lukaw3d) [inline src/components/ProjectDialog.tsx]: should this have `className='markdown-wrapper'` too?

File: src/components/ProjectDialog.tsx [hunk 2]
```diff
@@ -89,16 +95,20 @@ const ProjectDialog: React.FC<ProjectDialogProps> = ({
               </Carousel>
              <Grid container spacing={3} sx={{marginBottom: '32px', marginTop: '32px'}}>
                    <Grid item xs={12} md={6}> 
-                      <Typography
-                      sx={{color: '#445E77', letterSpacing: '-0.5px', marginBottom: '24px'}}
-                      >{project.description}</Typography>
+                      <Box
+                          sx={{color: '#445E77', letterSpacing: '-0.5px', marginBottom: '24px', fontFamily: "'Figtree Variable',sans-serif"}}
+                          >
+                             <ReactMarkdown components={{ a: ({ node, ...props }) => modifyLinkTarget(props.href, props.title, props.children) }}>
```
Comment (lukaw3d) [inline src/components/ProjectDialog.tsx:101]: Modifying markdown internals can have serious security issues. I think this should be moved to a separate pullrequest and need tests before accepting it.

File: src/components/ProjectDialog.tsx (hunk 1 above)
Comment (lukaw3d) [reply src/components/ProjectDialog.tsx]: No. markdown-wrapper actually clamps lines - i thought it affected link and code styling. Rename

File: src/index.css [hunk 3]
```diff
@@ -54,3 +57,15 @@ button:focus,
 button:focus-visible {
   outline: 4px auto #3431AC;
 }
+
+.markdown-wrapper {
+  overflow: hidden;
+  display: -webkit-box;
+  -webkit-line-clamp: 3;
+  line-clamp: 3;
+  -webkit-box-orient: vertical;
```
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/index.css:66]: Apparently just normal part of the out-of-spec clamping https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp#css

File: src/components/ProjectDialog.tsx (hunk 2 above)
Comment (kaja-osojnik) [reply src/components/ProjectDialog.tsx]: Moved to a separate pull request - https://github.com/oasisprotocol/playground/pull/37

**Pull request merged**